src/transformers/pipelines/__init__.py

```diff
@@ -727,28 +733,48 @@ def pipeline(
     if isinstance(model, Path):
         model = str(model)
 
+    if hub_kwargs["_commit_hash"] is None:
+        pretrained_model_name_or_path = None
+        if isinstance(config, str):
+            pretrained_model_name_or_path = config
+        elif config is None and isinstance(model, str):
+            pretrained_model_name_or_path = model
```

Comment:
try to get the repo. in different case before going to the next block of getting config, similar to the block in line 758